client/allocdir/fs_solaris.go

```diff
@@ -0,0 +1,26 @@
+package allocdir
+
+import (
+	"os"
+	"syscall"
+)
+
+// linkDir hardlinks src to dst. The src and dst must be on the same filesystem.
+func linkDir(src, dst string) error {
+	return syscall.Link(src, dst)
+}
+
+// unlinkDir removes a directory link.
+func unlinkDir(dir string) error {
+	return syscall.Unlink(dir)
+}
+
+// createSecretDir creates the secrets dir folder at the given path
+func createSecretDir(dir string) error {
```

Comment:
Potentially - right now this is "MVP" to get Nomad to compile in order to figure out what's broken and is a strict improvement on what is there now. Illumos does have `tmpfs` though, so likely this can be improved further.